PowerShellTools/Classification/ClassifierServices.cs

```diff
@@ -104,22 +105,51 @@ internal IEnumerable<TagInformation<ErrorTag>> TagErrorSpans(ITextBuffer buffer,
                 yield return new TagInformation<ErrorTag>(errorSpanStart, errorSpanLength, new ErrorTag("syntax error", parseError.Message));
             }
         }
+
+        /// <summary>
+        /// Returns tag information about the errors in the buffer provided. 
+        /// </summary>
+        /// <param name="buffer">The text buffer.</param>
+        /// <param name="spanStart">The start position of the current text span.</param>
+        /// <param name="errors">The parsed errors from out-proc.</param>
+        /// <returns>Error tags consumed by VS.</returns>
+        internal IEnumerable<TagInformation<ErrorTag>> TagErrorSpans(ITextBuffer buffer, int spanStart, IEnumerable<ParseErrorItem> errors)
+        {
+            var currentSnapshot = buffer.CurrentSnapshot;
+            foreach (var parseError in errors)
+            {
+                var errorSpanStart = parseError.ExtentStartOffset + spanStart;
+                var errorSpanLength = parseError.ExtentEndOffset - parseError.ExtentStartOffset;
+                if (errorSpanStart > currentSnapshot.Length || errorSpanStart + errorSpanLength > currentSnapshot.Length) continue;
+
+                if (errorSpanLength == 0)
+                {
+                    errorSpanLength = 1;
+                    if (errorSpanStart == currentSnapshot.Length)
+                    {
+                        errorSpanStart = currentSnapshot.Length - 1;
+                    }
+                }
+
+                yield return new TagInformation<ErrorTag>(errorSpanStart, errorSpanLength, new ErrorTag("syntax error", parseError.Message));

```

Comment:
Are we able to use PredefinedErrorTypeNames.SyntaxError instead of the string here?
[Inside MS.VS.Text.Adornments]